libs/live-wallet/src/store.ts

```diff
@@ -172,3 +210,83 @@ export const accountRawToAccountUserData = (raw: AccountRaw): AccountUserData =>
   }
   return { id, name, starredIds };
 };
+
+/**
+ * assuming local is up to date with wallet sync, we infer what could possibly be a new update to push.
+ */
+export function inferLocalToDistantDiff(
+  // accounts as they are in the global store
+  localAccounts: Account[],
+  // current local store
+  walletState: WalletState,
+): WalletSyncDiff {
+  let hasChanges = false;
+  const newNamesState = walletState.accountNames;
+  const currentState = walletState.wsState;
+
+  // check if the local accountNames have additions with upstream
+  if (currentState.data) {
+    const names = currentState.data.names;
+    for (const [id, name] of newNamesState) {
+      if (names[id] !== name) {
+        hasChanges = true;
+        break;
+      }
+    }
+  }
+
+  // let's figure out the new local accounts
+  const added: AccountDescriptor[] = [];
+  const distantServerAccountIds = new Set<string>(currentState.data?.accounts.map(a => a.id) || []);
+  for (const account of localAccounts) {
+    const id = account.id;
+    if (!distantServerAccountIds.has(id)) {
+      added.push({
+        id,
+        currencyId: account.currency.id,
+        freshAddress: account.freshAddress,
+        seedIdentifier: account.seedIdentifier,
+        derivationMode: account.derivationMode,
+        index: account.index,
+      });
+      hasChanges = true;
+    }
+  }
+
+  // let's figure out the locally deleted accounts that we will need to take into account
+  const removed: string[] = [];
+  const localAccountIds = new Set(localAccounts.map(a => a.id));
+  for (const id of distantServerAccountIds) {
+    if (!localAccountIds.has(id)) {
+      removed.push(id);
```

Comment:
TBD: how to manage the wsStateNonImportedAccountIds. maybe we shouldn't be considering the account ids of wsStateNonImportedAccountIds in this?